keps/sig-storage/20191031-storage-capacity-constraints-for-pod-scheduling.md

```diff
@@ -244,24 +243,44 @@ available, like health of a storage pool.
 
 ### Caching remaining capacity via the API server
 
-CSI defines the `GetCapacity` RPC call for the controller service to
-report information about the current capacity but it's not accessible
-in the Kubernetes scheduler. A new type `CSIStoragePool` type gets
-added to store the result of the `GetCapacity` calls for one storage
-pool in the API server and thus make it available to the scheduler. In
-the future it might get extended to also store other information like
-health of the pool.
-
-### Identifying storage pools
-
-The external-provisioner will be extended to handle the management
-of `CSIStoragePool` objects. For local storage in the node, there will
-be one such `CSIStoragePool` object per node and CSI driver on that
-node. For network attached storage, there will be one `CSIStoragePool`
-per CSI
-[Topology](https://github.com/container-storage-interface/spec/blob/4731db0e0bc53238b93850f43ab05d9355df0fd9/lib/go/csi/csi.pb.go#L1662-L1691)
-and driver. However, how the external-provisioner discovers these pools
-is currently open.
+The Kubernetes scheduler cannot talk directly to the CSI drivers to
+retrieve capacity information because CSI drivers typically only
+expose a local Unix domain socket and are not necessarily running on
+the same host as the scheduler(s).
+
+The key approach in this proposal for solving this is to gather
+capacity information, store it in the API server, and then use that
+information in the scheduler. That information then flows
+through different components:
+1. storage backend
+2. CSI driver
+3. Kubernetes-CSI sidecar
+4. API server
+5. Kubernetes scheduler
+
+The first two a driver specific. The sidecar will be provided by
```

Comment:
s/a/are?

Reply:
Urgh. Of course. I tend to loose letters when I get tired :cry: Fixed.